app/components/LocationPicker.tsx

```diff
@@ -16,13 +16,34 @@ const LocationPicker = ({
   onChangeLocation,
 }: LocationPickerProps) => {
   const [query, setQuery] = useState('');
+  const [searchResults, setResults] = useState<ArrayLike<any> | null | undefined>();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const newResults = orderBy(
+      filter(locations, ({ name }) =>
+        lowerCase(name).includes(lowerCase(query)),
+      ),
+      [({lastUsed}) => lastUsed !== undefined && lastUsed, ({ name }) => name.toLowerCase()],
```

Comment:
Shows the last used at top of the list before items are entered. Once user starts to empty string, filters down to search (with prev used (that match) still at top of list)